exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -36,12 +37,14 @@ import (
 )
 
 type exporter struct {
-	Config           *Config
-	logger           *zap.Logger
-	retryCount       int
-	collectorID      string
-	svcStructuredLog *cwlogs.Client
-	pusher           cwlogs.Pusher
+	Config                 *Config
+	logger                 *zap.Logger
+	retryCount             int
+	collectorID            string
+	svcStructuredLog       *cwlogs.Client
+	pusherMapLock          sync.Mutex
+	pusherOverride         cwlogs.Pusher
+	groupStreamToPusherMap map[string]map[string]cwlogs.Pusher
```

Comment:
I believe this should be abstracted out to cache type that looks like the following:

```
type PusherCache interface {
   // GetPusher will check to see if there already exists a pusher configured to write to 
   // the given stream and group, and that pusher will be returned.
   // If no pusher exists for a given stream and group, a new one is created, stored and returned
   GetPusher(stream, group string) cwlogs.Pusher
 
   // Shutdown will force each stored pusher to flush any data currently buffered
   // and then clear the existing cache. 
   Shutdown(ctx context.Context) error
}
```

Something to the above affect will help make this code a little more readable and make it clear of its intent.

Reply:
This is in line with what I was thinking [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12179#discussion_r917169285).

Reply:
I like this approach because it helps declare what the intent / usage here instead of a map of maps which I'm always cautious about.